astropy/io/fits/hdu/base.py

```diff
@@ -457,7 +457,7 @@ def block_iter(nbytes):
         hdu._data_size = size + _pad_length(size)
 
         # Checksums are not checked on invalid HDU types
-        if checksum and checksum != 'remove' and isinstance(hdu, _ValidHDU):
+        if isinstance(hdu, _ValidHDU):
```

Comment:
@MSeifert04 Yes. Exactly that's what I thought, `self._checksum` should really be used as such. Defeats the purpose. But I said all that considering that I don't have to make any changes to the script itself. 
I think I'll go and fix this in the script too.